shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm

```diff
@@ -365,6 +365,30 @@ - (CGFloat)contentsScale {
   return self.superview != nil ? self.superview.layer.contentsScale : 1.0;
 }
 
+/// Updates the nested stack of clip views that host the platform view.
+- (void)updatePathClipViewsWithPaths:(NSArray*)paths {
+  // Remove path clip views depending on the number of paths.
+  while (_pathClipViews.count > paths.count) {
+    NSView* view = _pathClipViews.lastObject;
+    [view removeFromSuperview];
+    [_pathClipViews removeLastObject];
+  }
+  // Otherwise, add path clip views to the end.
+  for (size_t i = _pathClipViews.count; i < paths.count; ++i) {
+    NSView* superView = _pathClipViews.count == 0 ? self : _pathClipViews.lastObject;
+    FlutterPathClipView* pathClipView = [[FlutterPathClipView alloc] initWithFrame:self.bounds];
+    [_pathClipViews addObject:pathClipView];
+    [superView addSubview:pathClipView];
+  }
+  // Update bounds and apply clip paths.
+  for (size_t i = 0; i < _pathClipViews.count; ++i) {
+    FlutterPathClipView* pathClipView = _pathClipViews[i];
+    pathClipView.frame = self.bounds;
+    [pathClipView maskToPath:(__bridge CGPathRef)[paths objectAtIndex:i]
+                  withOrigin:self.frame.origin];
```

Comment:
Note that `self.frame.origin` and `finalBoundingRect.origin` are the same point. See line 385.